.env.example.holesky

```diff
@@ -87,3 +87,6 @@ EIGENDA_PROXY_SERVICE_MANAGER_ADDR=0xD4A7E1Bd8015057293f0D0A557088c286942e84b
           
 # endpoint for S3 storage
 # EIGENDA_PROXY_S3_ENDPOINT
+
+# set to true to disable writing to EigenDA
+# EIGENDA_PROXY_READ_ONLY=false
```

Comment:
Thinking about this a bit more pragmatically - we have a LOT of knobs for this service with the env namespace only growing. Would it be better to process this internally using the `SignerPrivateKey` where we treat it as `readonly <-> SignerPrivateKey=""` or unset by the user? 

A lil hesitant to introduce more env flags, thoughts? 
cc - @samlaf, @bxue-l2

Reply:
Yeah I certainly don't like the fact that we have growing flags without obvious links between themselves. I don't really like the privateKey being empty turning readonly mode though, as that is very surprising. One thing we could do eventually is move to urfave v3 and use mutuallyExclusiveFlags (https://pkg.go.dev/github.com/urfave/cli/v3#MutuallyExclusiveFlags). So basically urfave would force user to either be in readonly mode, or to provide a signerPrivateKey, but can't have both.

Reply:
Regarding the UX problem of celo wanting to run in read-only mode and not having to provide a random signerPrivateKey, seems like the main source of the problem is in the eigenda-client? https://github.com/Layr-Labs/eigenda/blob/97dd2a3da09fd62ff3670afd96c2d402f1ca8453/api/clients/config.go#L55 seems like having a non-empty signerPrivateKey is enforced? Perhaps we would need a read-only eigenda-client?

Reply:
Yeah, I agree with all the concerns here. I would suggest having a separate refactor effort for the flags.

For comment on eigenda-client, I thought so too, but didn't want to touch `eigenda` code 😬  maybe I should've taken the initiative there which makes more sense. I can go ahead and put a PR there and see what corresponding changes would be good here

Reply:
reevaluating the approach. Maybe it makes more sense to make a retrieval client instead. Currently `EigenDAClient` only contains a dispersal client that has a separate definition for `RetrieveBlob` than the one for retrieval client that has a more complex/decentralized retrieval logic. 

thinking about read-only's UX, perhaps we want to simply create an `RetrievalClient` instead of an `EigenDAClient` and let `retrieveBlob` serve `Get` and return error for `Put`. But then, why don't the read-only user simply spin up a retrieval client? lmk what you guys think, can also bring this up in the sync

Reply:
We didn't have time to discuss that in the sync. In the current `EigenDAClient`, it looks we didn't put much consideration for clients that require read-only (like op-node or nitro node which only runs derivation). Instead of creating patch here and there, it is good to start from a high level design first, which I think Hope is pointing out.

On spinning up a retrieval client. In OP, the protocol requires there to be a proxy server, so the user can't spin up a client, the proxy has to. 

On refactor, we could create a client struct that wrap around both node and disperser client. What do people think

Reply:
Thanks Bowen, I think taking a step back and agree on high level design makes sense. I think the current PR might be a short term temporary solution if we are okay with this, but I guess no one is rushing to use this, so I'm okay to close this.

For the client struct, are you saying this client struct would live in `eigenda` or in `eigenda-proxy`? what's the reason to wrap around a node for a read only mode 🤔? what about waiting for contract storage changes for operator set and implement distributed retrieval for the retriever first?

Reply:
ignore my suggestion, it is probably too low level of a suggestion.

Reply:
okay, sorry if my questions were overwhelming. I'm more than happy to learn and understand what you think is best